### ISSUE TITLE
How to test the texture transfer model?

### ISSUE THREAD

**Issue by confifu** [Zhangjinso/PISE#26]
I want to test only the texture transfer model. It is mentioned that the model needs to be changed. How should I change it? What inputs are needed to test the texture transfer model?

**Comment by Zhangjinso** [comment]
As far as I can remember, you can do it with some steps.
1. You need to change [this line](https://github.com/Zhangjinso/PISE/blob/f9dfacb16db99420a5a51bf7e0fd2a632466b701/model/__init__.py#L9)  to 'model_file_name = "model." + model_name + "_model_transfer" ' to import the texture transfer model.
2. You also need to change [this line](https://github.com/Zhangjinso/PISE/blob/f9dfacb16db99420a5a51bf7e0fd2a632466b701/model/networks/__init__.py#L36) to ' filename = 'generator_texture_transfer' if filename is None else filename ' to import generator.
3. You need to change the dataloader and your test_pair.csv to input two reference images according to [this line](https://github.com/Zhangjinso/PISE/blob/f9dfacb16db99420a5a51bf7e0fd2a632466b701/model/painet_model_transfer.py#L137).
4. You need to uncomment [this line](https://github.com/Zhangjinso/PISE/blob/f9dfacb16db99420a5a51bf7e0fd2a632466b701/model/networks/generator_texture_transfer.py#L186) and modified the region index and 'alpha' (maybe change it to 1).
5. Then you can test it.

**Comment by happyday521** [comment]
@Zhangjinso Hi, in line 223 of generator_texture_transfer.py, what is the meaning of "if False"? Don't we need to execute indented codes under if？（line 223-265）Thanks！

**Comment by Zhangjinso** [comment]
Yes. The texture transfer model  does not used spatio-aware normalization module for the convenience of training. If you want to get better results, you can retrain the model with this.

**Comment by confifu** [comment]
I am getting this type of result for pose transfer as well as texture transfer. What am I doing wrong? 
![fashionMENTees_Tanksid0000730104_7additional_fashionWOMENBlouses_Shirtsid0000337203_2side_vis0](https://user-images.githubusercontent.com/35372948/122354484-aa925100-cf6e-11eb-98b0-2bf315899a6d.jpg)

**Comment by confifu** [comment]
Pose transfer results using the pre-trained checkpoint are not as good as shown in the paper. Were different checkpoints used to get the results shown in the paper?
![fashionWOMENDressesid0000713204_4full_fashionWOMENDressesid0000713204_3back_all](https://user-images.githubusercontent.com/35372948/122372553-6313c100-cf7e-11eb-9e65-6d54f17b1e18.jpg)
![fashionWOMENDressesid0000715402_4full_fashionWOMENDressesid0000715402_2side_all](https://user-images.githubusercontent.com/35372948/122372581-6a3acf00-cf7e-11eb-80e7-b92fa501683c.jpg)
![fashionWOMENJackets_Coatsid0000450202_1front_fashionWOMENJackets_Coatsid0000450202_7additional_all](https://user-images.githubusercontent.com/35372948/122372626-72930a00-cf7e-11eb-9852-12d1a3a0df18.jpg)
![fashionWOMENShortsid0000513803_4full_fashionWOMENShortsid0000513803_1front_all](https://user-images.githubusercontent.com/35372948/122372662-7c1c7200-cf7e-11eb-8afd-33a9cac71754.jpg)

**Comment by Zhangjinso** [comment]
You can refer to [this issue](https://github.com/Zhangjinso/PISE/issues/21#issuecomment-839797211).

**Comment by confifu** [comment]
What should be `self.input_SPL3` ? `SPL2_onehot` has shape `[8, 256, 256]`. Do we also need to input one hot vector in input_SPL3 as well? Do we only have to leave the region for which we want the texture transfer?

**Comment by confifu** [comment]
If anyone wants to do this,
To keep the pose same and only transfer texture, I did the following:
* Change the following line
 https://github.com/Zhangjinso/PISE/blob/f9dfacb16db99420a5a51bf7e0fd2a632466b701/data/base_dataset.py#L136
to `return {'P1': P1, 'BP1': BP1, 'P2': P1, 'BP2': BP1,'SPL1': SPL1_onehot, 'SPL2':SPL1_onehot, 'label_P2': SPL1,
                'P1_path': P1_name, 'P2_path': P1_name, 'P3' : P2, 'SPL3' : SPL2_onehot}`

* To change the `part` for which you want to transfer texture, edit the following line
https://github.com/Zhangjinso/PISE/blob/f9dfacb16db99420a5a51bf7e0fd2a632466b701/model/networks/generator_texture_transfer.py#L186
to `codes_vector[0,part,:] = (1-alpha)*codes_vector[0,part,:]+alpha*_codes_vector[0,part,:]`
set `alpha = 1` and `part` = 0 to 8

**Comment by mlyarthur** [comment]
![image](https://user-images.githubusercontent.com/29242717/128728854-4da6f7a1-410e-4769-8448-e63add321bbd.png)
After following your instruction for texture transfer model testing, I cannot get the desired result in the paper @eternitjl @Zhangjinso  Do you know what step I miss or make a mistake

**Comment by ghost** [comment]
> ![image](https://user-images.githubusercontent.com/29242717/128728854-4da6f7a1-410e-4769-8448-e63add321bbd.png) After following your instruction for texture transfer model testing, I cannot get the desired result in the paper @eternitjl @Zhangjinso Do you know what step I miss or make a mistake

@mlyarthur Hi，have you solved it?  I have a similar problem 
![image](https://user-images.githubusercontent.com/78525115/165905421-c16dcfcc-a822-4ac7-b5f0-01dc462248ea.png)
